mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/process/VoidCompletionCallback.java

```diff
@@ -6,16 +6,39 @@
  */
 package org.mule.runtime.extension.api.runtime.process;
 
-import org.mule.runtime.api.meta.model.ComponentModel;
-
 /**
- * This callback is how a {@link ComponentModel components} notifies that its execution is completed,
- * without producing any output.
+ * This callback is how non blocking routers notify the end of its execution when no result is produced.
+ * <p>
+ * In order to implement a void non blocking router, the method needs to:
  * <p>
- * When the execution of the component has finished, it has to notify the completion either by
- * invoking {@link #success()} or {@link #error(Throwable)} methods.
- * Only then will the component's execution be considered as completed and the next processor in the
+ * <ul>
+ * <li>Have a void return type</li>
+ * <li>Have an argument of this type</li>
+ * </ul>
+ * <p>
+ * When the non blocking operation has finished, it has to notify its completion either by
```

Comment:
this is confusing. Not clear if talking about an operation in a chain or the router itself. Would me much clearer if says something `when the router execution finishes`